src/k8s-engine/lima.ts

```diff
@@ -646,11 +661,11 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
   /**
    * Run `limactl` with the given arguments.
    */
-  protected async lima(...args: string[]): Promise<void> {
+  protected async lima(this: Readonly<this>, ...args: string[]): Promise<void> {
```

Comment:
Can you add a comment explaining:
1. What is this leading `this` argument? This function is still always invoked as `this.lima(...)` so is there really always an implicit `this` parameter in every class method?
2. Why do you need to indicate it's readonly, when it's not assigned to in this function, even indirectly?

Reply:
There are still references to `this.limactl` in `ensureArchitectureMatch` -- shouldn't these be off `LimaBackend`?

Reply:
See [TypeScript documentation on `this` parameters](https://www.typescriptlang.org/docs/handbook/2/classes.html#this-parameters) — the argument here is used to narrow the type of `this` within the function.  The containing class (`LimaBackend`) still must implement everything that the type asks for.

This does *not* make it a static method; it's just a member function where we narrow the type of `this` to be less than the original class.  (TypeScript has [static members](https://www.typescriptlang.org/docs/handbook/2/classes.html#static-members) for that.)

Here, we use [`Readonly<T>`](https://www.typescriptlang.org/docs/handbook/utility-types.html#readonlytype) to modify the type so none of the fields are assignable (to try to make it immutable).  That is, I'm _enforcing_ that it's not assigned to in this function (as an extra guard).

What I'm really after is something like C++'s [`const` member function](https://en.cppreference.com/w/cpp/language/member_functions#const-_and_volatile-qualified_member_functions), but that's not possible in TypeScript.

(I'll add some comments, probably near the top of the class.)